### ISSUE TITLE
 @trexminer trexminer mentioned this issue 5 days ago Unable to mine Conflux using T-REX ... Reason - Device does not have enough free memory to hold the DAG.. working fine with NBMINER

### ISSUE THREAD

**Issue by 21Noob** [trexminer/T-Rex#1122]
I am getting the same error...
"TREX: Can't find nonce with device [ID=0, GPU #0], Device does not have enough free memory to hold the DAG for epoch 65. Required: 5.08 GB, free: 5.02 GB."
I have 2 rtx 2060-
1.Gigabyte GeForce RTX 2060 OC edition.
2.Zotac Gaming RTX 2060
unable to mine conflux now with t-rex miner... able to do it before when i was having only 1 gpu.. tried everything almost.. increasing paging file , enable CPU graphic etc. but gives the same error...
Then i switched to NBminer and i have no issues... I can now mine conflux with NBMiner but not with T-REX...
It is not gpu fault... it has something to do with T-REX itself.
Please fix it or someone please help. 
I love to use T-REX......... Please fix it.

**Comment by grenelt** [comment]
@21Noob The code needs it's own space and the video driver also allocates memory. Even if NBminer is still working, it probably won't work long... I assume you're mining with windows what makes things worse - linux drivers doesn't allocate as much memory...

**Comment by 21Noob** [comment]
Yes I am on windows.... But just a day before it was working fine... and all of sudden it stopped on T-REX

**Comment by grenelt** [comment]
It stopped at the time where the DAG size grew by the next epoch.. For most of the other mineable coins there are calculators, but unfortunatley i can't find one for Conflux. But i found some estimations that Conflux' DAG grows by 1 to 1.5 GB per year from starting the mainnet in December 2020 with 4 GB. So my very rough guessing is that NBminer also will stop mining Conflux on 6GB gpus within the next four weeks, probably less.

**Comment by 21Noob** [comment]
I don't think so... It has stopped only after adding another graphic card to rig... and only in T-REX... other mining programs working fine...
Let's see what happens..

**Comment by mederhoo-script** [comment]
Try to Use NBminer instead, as T-Rex’s DAG calculation remains less optimized for certain GPUs

Device does not have enough free memory error.
Solutions:
You can Force DAG build mode 1 (reduces VRAM usage)
`--dag-build-mode 1`

Or Switch to NBminer for CFX (better memory optimization)
`./nbminer -a cfx -o stratum+tcp://cfx.pool.com:PORT -u WALLET`


**Tradeoff**: `--dag-build-mode 1` may reduce hashrate by ~1-2%.
